docs/development/module/off-chain-workers.md

```diff
@@ -0,0 +1,383 @@
+  }
+}
+```
+
+Then you would be able to sign transactions, and this is good for **DEVELOPMENT only**.
+
+### Add an App Subkey via CLI
+
+In a more realistic setting, after setting up your substrate node, you can add a new app subkey via
+the command line interface as followed.
+
+```bash
+# Generate a new account
+$ subkey -s generate
+
+# Submit a new key via RPC
+$ http localhost:9933 jsonrpc=2.0 id=1 method=author_insertKey params:='["<YourKeyTypeId>", "<YourSeedPhrase>", "YourPublicKey"]'
+```
+
+A new key is now added in the local keystore.
+
+## Signed Transactions
+
+Now you are ready to to make a signed transaction from the off-chain worker as shown in the following.
+
+**runtime/src/my_runtime.rs**
+
+```rust
+decl_module! {
+  pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+    // ...
+    // ...
+    // This is your regular on-chain extrinsics
+    pub fn onchain_callback(origin, _block: T::BlockNumber, input: Vec<u8>) -> dispatch::Result {
+      let who = ensure_signed(origin)?;
+      debug::info!("{:?}", core::str::from_utf8(&input).unwrap());
+      Ok(())
+    }
+
+    fn offchain_worker(block: T::BlockNumber) {
+      // Here we specify the function to be callback on-chain in next block import
+      let call = Call::onchain_callback(block, b"hello world!".to_vec());
+      T::SubmitSignedTransaction::submit_signed(call);
+    }
+  }
+}
+```
+
+Here, after having defined the on-chain callback function, in the off-chain worker you can specify
+that function to be called back in the next block import phase. You then submit a signed transaction
+to the node.
+
+If you look at the implementation of `fn system::offchain::submit_signed ()` in [Substrate codebase](https://github.com/paritytech/substrate), you would realized it is calling the on-chain callback for each of the key in local keystore.
+But since you only have one key in the local keystore now, you are calling the function only once back on-chain.
+
+## Unsigned Transactions
+
+With the following code, you are able to send an unsigned transaction back on-chain.
+
+**runtime/src/my_runtime.rs**
+
+```rust
+decl_module! {
+  pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+    // ...
+    // ...
+    // This is your regular on-chain extrinsics
+    pub fn onchain_callback(_origin, _block: T::BlockNumber, input: Vec<u8>) -> dispatch::Result {
+      debug::info!("{:?}", core::str::from_utf8(&input).unwrap());
+      Ok(())
+    }
+
+    fn offchain_worker(block: T::BlockNumber) {
+      // Here we specify the function to be callback on-chain in next block import
+      let call = Call::onchain_callback(block, b"hello world!".to_vec());
+      T::SubmitUnsignedTransaction::submit_unsigned(call);
+    }
+  }
+}
+```
+
+By default, all unsigned transactions are treated as invalid transactions. You need to add the following
+code piece in `my_runtime.rs` to whitelist some on-chain functions that are allowed to be called with
```

Comment:
```suggestion
code piece in `my_runtime.rs` to explicitly allow some on-chain functions to be called with
```

Stealing this suggestion from Fuchsia docs https://fuchsia.dev/fuchsia-src/contribute/best-practices/respectful_code#what_are_examples_of_terminology_to_be_avoided